include/electionguard/chaum_pedersen.hpp

```diff
@@ -6,10 +6,80 @@
 #include "group.hpp"
 #include "precompute_buffers.hpp"
 
+#include <map>
 #include <memory>
 
 namespace electionguard
 {
+    /// <summary>
+    /// Validation result for a zero knowledge proof
+    /// </summary>
+    struct ValidationResult {
+        bool isValid;
+        std::vector<std::string> messages;
+    };
+
+    /// <summary>
+    /// A generic zero knowledge proof struct that can encapsulate
+    /// any of the individual proofs used for ballot correctness in electionguard
+    /// </summary>
+    struct ZeroKnowledgeProof {
```

Comment:
created this to capture the structure of a zkp that we use. it is only added to the range proof for now but should replace the internal structure of the disjunctive as well